assets/js/settings/shared/index.js

```diff
@@ -1,3 +1,32 @@
-export { getSetting } from './get-setting';
-export { setSetting } from './set-setting';
 export * from './default-constants';
+export { setSetting } from './set-setting';
+
+/**
+ * Internal dependencies
+ */
+import { getSetting } from './get-setting';
+import compareVersions from 'compare-versions';
+
+/**
+ * Note: this attempts to coerce the wp_version to a semver for comparison
+ * This will result in dropping any beta/rc values.
+ *
+ * `5.3-beta1-4252` would get converted to `5.3.0-rc.4252`
+ * `5.3-beta1` would get converted to `5.3.0-rc`.
+ * `5.3` would not be touched.
+ *
+ * For the purpose of these comparisons all pre-release versions are normalized
+ * to `rc`.
+ */
+export const compareWithWpVersion = ( version, operator ) => {
+	let replacement = getSetting( 'wp_version', '' ).replace(
+		/-[a-zA-Z0-9]*[\-]*/,
+		'.0-rc.'
+	);
+	replacement = replacement.endsWith( '.' )
+		? replacement.substring( 0, replacement.length - 1 )
+		: replacement;
+	return compareVersions.compare( version, replacement, operator );
```

Comment:
Not sure why, but I expected this function to work the other way around (so `version` and `replacement` to be switched).

```suggestion
	return compareVersions.compare( replacement, version, operator );
```

In the current code, if we want to test if WP is higher than 5.3, we have to write the _lower than_ sign:
```ES6
if ( compareWithWpVersion( '5.3.0', '<' ) ) {
	// version is higher than 5.3.0
}
```
I find it counter-intuitive. But if it's just me, I'm ok leaving it as-is.

Reply:
Hmm, it's intentionally written as you would if you were doing the conditional:

```js
// compareWithWpVersion( '5.3.0', '<' ) is the equivalent of.
if ( '5.3.0' < wpVersion ) {
 // do stuff
}
```

So you're feeding the version (left side of operator) that you want to check against the wpVersion (which would be on the right side of the operator).  This is also following the pattern of how the underlying `compareVersions` library works (where comparison is left to right using the operator).